### PR TITLE
Flux Monitor: Improve AddJob(..) concurrency

### PR DESCRIPTION
**Pull request by felder-cl**
Improve concurrency to minimize startup time by separating the methods which rely on variables protected by the CSP model.

This preserves previous expected behavior before introducing the fix #e1d9a36.

https://www.pivotaltracker.com/story/show/171020328

### CODE REVIEW

File: core/services/flux_monitor.go [hunk 1]
```diff
@@ -155,31 +184,32 @@ func connectCheckers(ctx context.Context, jobMap map[string][]DeviationChecker,
...
-		logger.Debugw("Adding job to flux monitor",
-			"job", job.ID.String(),
-			"initr", initr.ID,
+func (fm *concreteFluxMonitor) addAction(
+	ctx context.Context,
+	connected bool,
+	jobMap map[string][]DeviationChecker,
+	store *store.Store,
+	jobSpecID string,
+	checkers ...DeviationChecker,
```
Comment (j16r) [inline core/services/flux_monitor.go]: This splat operator is not being used.

File: core/services/flux_monitor.go [hunk 2]
```diff
@@ -73,12 +74,20 @@ func (fm *concreteFluxMonitor) Start() error {
...
-		merr = multierr.Combine(merr, err)
+		go func(j *models.JobSpec) {
+			errChan <- fm.AddJob(*j)
+		}(j)
+		count++
 		return true
 	}, models.InitiatorFluxMonitor)
+
+	var merr error
+	for i := 0; i < count; i++ {
```
Comment (j16r) [inline core/services/flux_monitor.go]: This is just making a list of the error returned by `Jobs` (which is just a single error) with the size of the number of jobs - is that really what you want? e.g: If you can't connect to the DB, you'll get:

    ["error failed to connect", "error failed to connect", "error failed to connect", ...]

File: core/services/flux_monitor.go [hunk 3]
```diff
@@ -73,12 +74,20 @@ func (fm *concreteFluxMonitor) Start() error {
...
 	go fm.actionConsumer(fm.ctx)
 
-	var merr error
+	count := 0
+	errChan := make(chan error)
 	err := fm.store.Jobs(func(j *models.JobSpec) bool {
-		err := fm.AddJob(*j)
-		merr = multierr.Combine(merr, err)
+		go func(j *models.JobSpec) {
+			errChan <- fm.AddJob(*j)
```
Comment (j16r) [inline core/services/flux_monitor.go]: `errChan` has no buffer, and is never read from, what is it for?

File: core/services/flux_monitor.go (hunk 2 above)
Comment (felder-cl) [reply core/services/flux_monitor.go]: The error here should have been taken from the error channel mentioned above

File: core/services/flux_monitor.go (hunk 3 above)
Comment (felder-cl) [reply core/services/flux_monitor.go]: It was for the case mentioned above, thanks for catching this one.

This code should mirror what was implemented before e1d9a36

File: core/services/flux_monitor.go (hunk 1 above)
Comment (felder-cl) [reply core/services/flux_monitor.go]: Lets agree to disagree about this, but changed this for the PR

**Pull request merged**